docs/deprecated.md

```diff
@@ -110,6 +111,19 @@ The table below provides an overview of the current status of deprecated feature
 | Removed    | [`--run` flag on `docker commit`](#--run-flag-on-docker-commit)                                                                    | v0.10      | v1.13  |
 | Removed    | [Three arguments form in `docker import`](#three-arguments-form-in-docker-import)                                                  | v0.6.7     | v1.12  |
 
+### `Container` and `ContainerConfig` fields in Image inspect
+
+**Deprecated in Release: v25.0**
```

Comment:
Did we forget mentioning these as deprecated for the v25 release? Wondering if that means we need to wait for v27.0 to remove them, or did we document it in the API changes at the time?

Reply:
Yes, we forgot to put it there, but we mentioned it in `version-history.md` 🙈 

> The Container and ContainerConfig fields in the GET /images/{name}/json response are deprecated and will no longer be included in API v1.45.

But I'm fine with waiting for v27.0 for the removal, especially since the docker-py with the fixed tests is not released yet.

Reply:
Arf, I somehow missed your last comments. Some thoughts (_maybe_ we don't have to postpone the removal);

- The API changes document is the "canonical" place for changes in the engine API, _including_ deprecations
- I _thought_ we always included a mention like "for changes and deprecations in the API refer to `<link to API changes>`"; if we don't, we should fix that.
- :point_up: perhaps we should also do a cross-reference in this `deprecated` page, to do the same ("for an overview of changes in API ...")

That said; I don't think it necessarily _hurts_ to call out / repeat important changes here, but we may not _have to_ for all of them.


@vvoland @dvdksn WDYT?

Reply:
FWIW; for some of these, it would at least make things easier as well, because the API is versioned so if you use the "older API version", nothing changes, but yeah, in some cases that may affect output of `docker` CLI commands.

`docker inspect` was meant to show the underlying "low-level" information, which would still be "accurate" here, but that underlying information now may no longer have these fields (or not set).

Reply:
We link to both the `deprecated.md` and `version-history.md` from the release note:

<img width="820" alt="image" src="https://github.com/docker/cli/assets/35727626/997ee51e-0560-42ec-830e-361bea43a852">


I don't think it hurts to also cross-reference between these two.

If `version-history.md` is canonical then we should be safe to remove in 26.0 since this was called out (and linked from the release note). It just wasn't called out everywhere.

<img width="819" alt="image" src="https://github.com/docker/cli/assets/35727626/fab632b1-1db5-4be4-951d-898aad786446">

Reply:
Updated the removal notice to v26.